magento/pwa-studio
# [PB-PWA-1936]: Sorting product listing by custom attribute

## Description

- Gets available sort methods from `products` query
- As of writing, when sorting products by a new `custom_attribute` in ASC order, the returned products from the query will list products that do not have that attribute first.
- Had to hardcode some sorting values as they were coming from the component's `defaultProps` in `productSort.js` and there is no way to merge incoming + default props.

## Related Issue

Closes [#1936](https://jira.corp.magento.com/browse/PWA-1936)

## Acceptance

### Verification Stakeholders

### Specification

## Verification Steps
- Visit [Bottoms-page](https://pr-3619.pwa-venia.com/venia-bottoms/venia-pants.html?page=1)
- Check if sort options are alphabetically ordered
- Check that custom attribute `Fit` is in the list
- Verify that right click does not select sorting
- Sort list by clicking the `Fit` custom attribute (Products that contain this custom attribute: `Clara Wide Leg Pants (Fit: Baggy), Gloria Palazzo Pants (Fit: Baggy), Bella Eyelet Capris (Fit: Fit), Selena Pants (Fit: fit)`
- Baggy < Fit so Baggy attributes should show up first when ordering by ascending
- Check same steps above using different language/store. To test this install venia language pack extension by running `yarn venia add -D @magento/venia-sample-language-packs`, run application and select `French Store View` on header. 

#### Test scenario(s) for direct fix/feature

#### Test scenario(s) for any existing impacted features/areas

Verify user is able to sort data after applying filters on category page
Verify user is able to sort data after applying filters on search page
